packages/hapi/modules/plugin/plugin.js

```diff
@@ -38,16 +37,13 @@ const createPlugin = ({ options } = {}) => {
   const plugin = {
     name: pkg.name,
     version: pkg.version,
-    register(server, options, onDone = () => null) {
-      const areServerEventsSupported = Boolean(
-        server.version && semiver(server.version, '17.0.x')
-      );
-      const onRequestHandler = (request, h) => {
+    register(server, o, done) {
+      server.ext('onRequest', (request, h) => {
         request.promster = { start: process.hrtime() };
         return h.continue;
-      };
+      });
 
-      const onResponseHandler = request => {
+      server.ext('onPreResponse', (request, h) => {
```

Comment:
Please check again with `master`. I now compare versions of Hapi to pick this API if < 17.0.x.